builder/amazon/common/step_get_password.go

```diff
@@ -52,7 +53,7 @@ func (s *StepGetPassword) Run(ctx context.Context, state multistep.StateBag) mul
 		ui.Message(
 			"It is normal for this process to take up to 15 minutes,\n" +
 				"but it usually takes around 5. Please wait.")
-		password, err = s.waitForPassword(state, cancel)
+		password, err = s.waitForPassword(state, cancel, ctx)
```

Comment:
Here, we can remove the cancel chan, the context will be cancelled and we don't need to check for `multistep.StateCancelled` down on line 83.
Line 83 we should rather check for the context to be `Done`.

Reply:
Also it is a good practice to put the context as the first argument.